src/Kestrel.Core/Internal/ConnectionHandler.cs

```diff
@@ -24,38 +28,73 @@ public ConnectionHandler(ListenOptions listenOptions, ServiceContext serviceCont
             _application = application;
         }
 
-        public IConnectionContext OnConnection(IConnectionInformation connectionInfo)
+        public void OnConnection(IFeatureCollection features)
         {
-            var inputPipe = connectionInfo.PipeFactory.Create(GetInputPipeOptions(connectionInfo.InputWriterScheduler));
-            var outputPipe = connectionInfo.PipeFactory.Create(GetOutputPipeOptions(connectionInfo.OutputReaderScheduler));
+            var connectionContext = new DefaultConnectionContext(features);
+
+            var transportFeature = connectionContext.Features.Get<IConnectionTransportFeature>();
```

Comment:
I though `DefaultConnectionContext.ConnectionTransportFeature` was already public. Didn't read that carefully enough. I guess there will only be one ConnectionHandler, so maybe it's not a big deal that it's ugly and hard-to-discover.